src/components/distributions/editor/TextForm/TextInput.js

```diff
@@ -4,31 +4,70 @@ import $ from 'jquery'
 import {EditorState, Editor, ContentState, Modifier, CompositeDecorator} from 'draft-js'
 
 import {isData, formatData} from 'lib/guesstimator/formatter/formatters/Data'
-import {getFactParams, addText, addSuggestionToEditorState, STATIC_DECORATOR, STATIC_DECORATOR_LIST} from 'lib/factParser'
+import {getFactParams, addText, addSuggestionToEditorState, findWithRegex, FACT_DECORATOR_LIST} from 'lib/factParser'
+
+const ValidInput = props => <span {...props} className='valid input'>{props.children}</span>
+const ErrorInput = props => <span {...props} className='error input'>{props.children}</span>
 
 export default class TextInput extends Component{
   displayName: 'Guesstimate-TextInput'
 
   state = {
-    editorState: EditorState.createWithContent(ContentState.createFromText(this.props.value || ''), new CompositeDecorator(STATIC_DECORATOR_LIST)),
+    editorState: EditorState.createWithContent(
+      ContentState.createFromText(this.props.value || ''),
+      new CompositeDecorator(this.decoratorList()),
+    ),
     suggestion: {
       text: '',
       suffix: '',
     },
+    extraDecorators: [],
+    decoratorsUpToDate: false,
   }
 
   static propTypes = {
     value: PropTypes.string,
   }
 
+  decoratorList() {
```

Comment:
Not particularly, but as they're already merged to master, it doesn't seem like a bad thing